### PR TITLE
FIX: NTP improvements and Tweaks enhancements

### PR DESCRIPTION
**Pull request by tGecko**
**Network time:**
Add some feedback for the user if getting network time was successful on boot
Wait for IP address before starting to sync the time (relevant on boot)
Reduce ping timeout and maximum amount of attempts for sync_time

**Tweaks:**
Add toggle to disable network services while in game, set it to on by default
Add diagnostics menu with toggle to enable logging

**General**
Add logging script to be used with shell script for consistent log name and timestamp format
may be expanded to a more general "tools" script with things like get_flag or is_running

### CODE REVIEW

**Comment by Aemiii91** [comment]
What a great idea! 😁

File: src/tweaks/appstate.h [hunk 1]
```diff
@@ -67,6 +67,7 @@ static List _menu_battery_percentage;
 static List _menu_advanced;
 static List _menu_reset_settings;
 static List _menu_tools;
+static List _menu_diagnostics;
 
 void menu_free_all(void)
 {
```
Comment (Aemiii91) [inline src/tweaks/appstate.h:73]: You need to add `list_free(&_menu_diagnostics)` for the new menu here

File: src/tweaks/menus.h [hunk 2]
```diff
@@ -490,10 +480,26 @@ void menu_resetSettings(void *_)
     header_changed = true;
 }
 
+void menu_diagnostics(void *_)
+{
+    if (!_menu_diagnostics._created) {
+        _menu_diagnostics = list_create(1, LIST_SMALL);
```
Comment (Aemiii91) [inline src/tweaks/menus.h:486]: Please use the new `list_create_with_title` (it's cleaner)

File: static/build/.tmp_update/runtime.sh [hunk 3]
```diff
@@ -344,6 +357,9 @@ launch_game() {
     # Free memory
     $sysdir/bin/freemma
 
+    # Reset networking
+    touch /tmp/network_changed
```
Comment (Aemiii91) [inline static/build/.tmp_update/runtime.sh:361]: This should be behind a condition I think - and actually test if any network services are enabled - and whether `keepAlive` is disabled

File: static/build/.tmp_update/script/network/update_networking.sh [hunk 4]
```diff
@@ -424,45 +424,64 @@ check_hotspotstate() {
 # This will work but it will not export the TZ var across all opens shells so you may find the hwclock (and clock app, retroarch time etc) are correct but terminal time is not.
 # It does set TZ on the tty that Main is running in so this is ok
 
-sync_time() {
-    if [ -f "$sysdir/config/.ntpState" ] && wifi_enabled; then
+check_ntpstate() {
+    if flag_enabled ntpState && wifi_enabled && [ ! -f "$sysdir/config/.hotspotState" ]; then
+        set_tzid
+        if [ -f /tmp/ntp_synced ] && [ ! "$is_booting" -eq 1 ]; then
```
Comment (Aemiii91) [inline static/build/.tmp_update/script/network/update_networking.sh:430]: Why is the second condition necessary?

Comment (Aemiii91) [reply static/build/.tmp_update/script/network/update_networking.sh:430]: Also, "NOT EQ 1" should just be "EQ 0"

File: static/build/.tmp_update/script/network/update_networking.sh [hunk 5]
```diff
@@ -424,45 +424,64 @@ check_hotspotstate() {
...
+                if [ $attempts -eq $max_attempts ]; then
+                    log "NTPwait: Ran out of time before we could sync, stopping."
+                    ret_val=1
+                    break
+                fi
+                sleep 1
+            done
         fi
     fi
+    touch /tmp/ntp_synced
```
Comment (Aemiii91) [inline static/build/.tmp_update/script/network/update_networking.sh:478]: No. NTP is not always synced when getting to here.

**Review by Aemiii91 — CHANGES_REQUESTED** [review]

File: static/build/.tmp_update/script/network/update_networking.sh (hunk 4 above)
Comment (tGecko) [reply static/build/.tmp_update/script/network/update_networking.sh:430]: the check for is_booting allows the sync on boot

Comment (Aemiii91) [reply static/build/.tmp_update/script/network/update_networking.sh:430]: `/tmp/ntp_synced` already doesn't exist on boot

File: static/build/.tmp_update/runtime.sh (hunk 3 above)
Comment (tGecko) [reply static/build/.tmp_update/runtime.sh:361]: all these checks happen in update_networking check() (which is what's eventually called by touching this flag),  no?

File: static/build/.tmp_update/script/network/update_networking.sh (hunk 4 above)
Comment (tGecko) [reply static/build/.tmp_update/script/network/update_networking.sh:430]: gotcha 👍

File: static/build/.tmp_update/runtime.sh (hunk 3 above)
Comment (Aemiii91) [reply static/build/.tmp_update/runtime.sh:361]: The `network_changed` flag was added to avoid doing the checks all the time - this negates that

**Review by Aemiii91 — APPROVED** [review]

**Pull request merged**